docs/proposals/08-peering.md

```diff
@@ -0,0 +1,168 @@
+
+### The `NetworkPeeringRequest`type
+
+```yaml
+apiVersion: networking.api.onmetal.de/v1alpha1
+kind: NetworkPeeringRequest
+metadata:
+  namespace: customer-manuel
+  name: network-sample-request
+spec:
+  localNetworkRef:
+    name: manuels-network
+  remoteNetworkRef:
+    name: mysupernetwork
+    namespace: customer-andre #optional could be another network of the same namespace
+status:
+  ttl: 2023-02-15T19:00:00Z # omit if empty
+  state: -> pending because missing match
+         -> failed overlapping ip range
+         -> success when found a match and peered
+```
+
+* `localNetworkRef`: reference to the network that the creator of this `NetworkPeeringRequest` owns and intends to peer, needs to reside in the same namespace as the referenced `Network`object
+* `remoteNetworkRef`: reference to the network that should be peered with the one referenced in the `localNetworkRef`, namespace can be omited if this `Network`object resides in the same namespace as the `NetworkPeeringRequest``
+* `status.ttl`: ISO timestamp that reflects the point in time when this peering request is being deleted if the `status.state`is still `pending` or `failed` before. If `status.state`is `success`, this will be ignored and the object will not be deleted
+* `status.state`: can be either `pending`, `success` or `failed`. State is `pending` as long as there is no matching `NetworkPeeringRequest` present. State is `success`when there is a matching `NetworkPeeringRequest` present. State is `failed` when certain parameters stop the matching from succeeding, like e.g. overlapping IP ranges of the two referenced networks
+
+### The updated `Network` type
+
+```yaml
+apiVersion: networking.api.onmetal.de/v1alpha1
+kind: Network
+metadata:
+  namespace: customer-manuel
+  name: manuels-network
+status:
+  peeredNetworks:
+    - name: mysupernetwork
+      namespace: customer-andre
+```
+
+* `status.peeredNetworks`: is a list of successfully peered `Networks`
+
+### Example
+
+Let's suppose Manuel wants to peer his `Network` `manuels-network` with André's `Network` `mysupernetwork`. Let's also assume that Manuel's namespace is `customer-manuel` and André's namespace is called `customer-andre`. With that we get the following two `NetworkPeeringRequest`objects reflecting a successful peering of the two `Network`objects presented at the end.
```

Comment:
Please remove your own names to keep this generic. Also, remove the usage of the word 'customer' - this is about the API and namespaces, not about customers (that will eventually be there for sure, but customer / IAM can only exist if we respect k8s namespaces and boundaries correctly).